k2/csrc/array_ops_test.cu

```diff
@@ -1295,4 +1296,81 @@ TEST(OpsTest, Array2IndexTest) {
   }
 }
 
+template <typename T>
+static void Array1SortTestSimple() {
+  std::vector<T> data = {3, 2, 5, 1};
+  for (auto &context : {GetCpuContext(), GetCudaContext()}) {
+    {
+      // with index map
+      Array1<T> array(context, data);
+      Array1<int32_t> index_map;
+      Sort(&array, &index_map);
+      CheckArrayData(array, std::vector<T>{1, 2, 3, 5});
+      CheckArrayData(index_map, std::vector<int32_t>{3, 1, 0, 2});
+    }
+
+    {
+      // without index map
+      Array1<T> array(context, data);
+      Sort(&array);
+      CheckArrayData(array, std::vector<T>{1, 2, 3, 5});
+    }
+  }
+}
+
+template <typename T>
+static void Array1SortTestEmpty() {
+  for (auto &context : {GetCpuContext(), GetCudaContext()}) {
+    Array1<T> array(context, 0);
+    Array1<int32_t> index_map;
+    Sort(&array, &index_map);
+    EXPECT_EQ(array.Dim(), 0);
+    EXPECT_EQ(index_map.Dim(), 0);
+  }
+}
+
+template <typename T>
+static void Array1SortTestRandom() {
+  int32_t num = RandIntGenerator()(0, 10000);
+  std::vector<T> data(num);
+  std::iota(data.begin(), data.end(), 0);
+  std::vector<T> saved_data = data;
+
+  std::random_shuffle(data.begin(), data.end());
```

Comment:
just call `RandUniformArray1` would be fine., no `iota`, no `shuffle`

Reply:
And you can check ans with `is_sorted`

Reply:
BTW, iota approach has a disadvantage that it never generates duplicate values

Reply:
Was not aware of `std::is_sort`,  thanks.